cpp/src/arrow/engine/substrait/options.h

```diff
@@ -67,6 +69,16 @@ using NamedTableProvider =
     std::function<Result<compute::Declaration>(const std::vector<std::string>&)>;
 static NamedTableProvider kDefaultNamedTableProvider;
 
+using NamedTapKindMapper = std::function<Result<compute::Declaration>(
+    const std::string&, std::vector<compute::Declaration::Input>,
+    std::shared_ptr<compute::ExecNodeOptions>)>;
+static NamedTapKindMapper kDefaultNamedTapKindMapper =
```

Comment:
Do you think we will want to make the default named tap mapper (or the default named table provider for that matter) configurable?

Reply:
I made it configurable.